extensions/amp-mustache/0.2/amp-mustache.js

```diff
@@ -24,14 +24,20 @@ import mustache from '../../../third_party/mustache/mustache';
 
 const TAG = 'amp-mustache';
 
+/**
+ * @typedef {BaseTemplate$$module$src$service$template_impl}
+ */
+let BaseTemplateDef;
+
 /**
  * Implements an AMP template for Mustache.js.
  * See {@link https://github.com/janl/mustache.js/}.
  *
  * @private Visible for testing.
- * @extends {BaseTemplate$$module$src$service$template_impl}
+ * @extends {BaseTemplateDef}
+ * @suppress {checkTypes}
  */
-export class AmpMustache extends AMP.BaseTemplate {
+export class AmpMustache extends BaseTemplateDef {
```

Comment:
This will break the inheritance chain. It must be `AMP.BaseTemplate` at runtime.

Reply:
Gotcha. I'm going to revert all changes to this file. The only problem that remains is to satisfy the type checker. Using `@suppress {checkTypes, reportUnknownTypes}` did not work. Still looking for an alternative solution.

```js
extensions/amp-mustache/0.1/amp-mustache.js:39: ERROR - Bad type annotation. Unknown type AMP.BaseTemplate
export class AmpMustache extends AMP.BaseTemplate {
                                 ^

extensions/amp-mustache/0.2/amp-mustache.js:35: ERROR - Bad type annotation. Unknown type AMP.BaseTemplate
export class AmpMustache extends AMP.BaseTemplate {
                                 ^
```

Reply:
@erwinmombay Any ideas for how to make `AMP.BaseTemplate` visible to `AmpMustache` without importing `BaseTemplate`?

Reply:
Can't you typecast `/** @type {BaseTemplate$$module$src$service$template_impl} */(AMP.BaseTemplate)`. This should work the same as extending from `AMP.BaseElement`, which is also not defined in extensions.

Reply:
Yep, typedef'ing `AMP.BaseTemplate` worked. Done in https://github.com/ampproject/amphtml/pull/21618/commits/9268ec8e2011790c6d134bf1804978dd86a0acdd.